WebAPI/Controllers/StudentController.cs

```diff
@@ -180,5 +179,61 @@ public async Task<IActionResult> AddNewBlock([FromBody]AddNewBlockModel newBlock
             }
             return Ok();
         }
+
+        [HttpPut("editBlock")]
+        public async Task<IActionResult> EditBlock([FromBody]AddNewBlockModel newBlockModel)
+        {
+            var student = await _studentService.FindByIdAsync(newBlockModel.User.Student.Id);
+
+            if (student == null)
+            {
+                return ErrorResponse($"Student with id: {newBlockModel.User.Student.Id} does not exist.");
+            }
+
+            if (student.Timetable == null)
+            {
+                return ErrorResponse($"Timetable for student with id: {student.Id} does not exist.");
+            }
+
+            Block block = new Block();
+            TimetableBlock timetableBlock = newBlockModel.TimetableBlock;
+            Course course = await _courseService.FindByNameAsync(timetableBlock.CourseName);
+            if (course == null)
```

Comment:
inside this if, should be just:
1. new course
2. AddAsync(course)
else is not needed and 211-221 could be deleted

Reply:
May be som model should be called responsible for adding new course (it will also call TAPI)